### PR TITLE
Hail backend frequency filter

### PR DESCRIPTION
**Pull request by hanars**

### CODE REVIEW

File: hail_search/hail_search_query.py [hunk 1]
```diff
@@ -250,6 +250,40 @@ def _missing_entry(entry):
...
+            if freqs.get('hh') is not None:
+                hom_field = pop_config['hom']
+                hemi_field = pop_config['hemi']
+                if hom_field:
+                    pop_filters.append(pop_expr[hom_field] <= freqs['hh'])
+                if hemi_field:
+                    pop_filters.append(pop_expr[hemi_field] <= freqs['hh'])
+
+            if pop_filters is not None:
+                pop_filter = pop_filters[0]
```
Comment (ShifaSZ) [inline hail_search/hail_search_query.py:282]: The condition for the if statement should be `len(pop_filters) > 0`. Otherwise, `pop_filters[0]` can be out of range error.

File: hail_search/test_search.py [hunk 2]
```diff
@@ -117,6 +117,36 @@ async def test_multi_project_search(self):
             sample_data=MULTI_PROJECT_SAMPLE_DATA,
         )
 
+    async def test_frequency_filter(self):
+        await self._assert_expected_search(
+            [VARIANT1, VARIANT4], frequencies={'seqr': {'af': 0.2}}, omit_sample_type='SV_WES',
```
Comment (ShifaSZ) [inline hail_search/test_search.py:122]: It is better to have a test for a population that has no frequencies or filters on a frequency other than 'af', 'ac', and 'hh'. It is the corner case that the `pop_filters` above can be empty.

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: hail_search/hail_search_query.py (hunk 1 above)
Comment (hanars) [reply hail_search/hail_search_query.py:282]: `len(None)` throws an error though. But yes, this could be cleaned up

File: hail_search/test_search.py (hunk 2 above)
Comment (hanars) [reply hail_search/test_search.py:122]: every other test in the file tests what happens when theres no frequency filter

Comment (hanars) [reply hail_search/test_search.py:122]: oh you mean has values that are all None. Yes, I can add a test

**Review by ShifaSZ — APPROVED** [review]
